### PR TITLE
convert for DataFrameRow

### PR DESCRIPTION
**Pull request by bkamins**
PR following discussion in #1564.

### CODE REVIEW

File: src/dataframerow/dataframerow.jl [hunk 1]
```diff
@@ -57,7 +57,18 @@ function Base.iterate(r::DataFrameRow, st)
...
+function Base.convert(::Type{Array}, r::DataFrameRow)
+    Base.depwarn("Conversion of `DataFrameRow` to a `Matrix` is deprecated. " *
+                 "`Vector` will be returned in the future.", :convert)
+    convert(Array, parent(r)[row(r),:])
+end
+Base.convert(::Type{Vector}, dfr::DataFrameRow) =
+    [dfr[i] for i in 1:length(dfr)]
+Base.convert(::Type{Vector{T}}, dfr::DataFrameRow) where T =
+    T[dfr[i] for i in 1:length(dfr)]
+Base.Vector(dfr::DataFrameRow) = Base.convert(Vector, df)
```
Comment (nalimilan) [inline src/dataframerow/dataframerow.jl]: No need for `Base.`.

File: test/dataframerow.jl [hunk 2]
```diff
@@ -100,5 +100,12 @@ module TestDataFrameRow
     io = IOBuffer()
     show(io, DataFrameRow(df, 1))
     @test String(take!(io)) == "DataFrameRow (row 1)\na  \nb  1"
+
+    df = DataFrame(a=[1, missing], b=[2.0, 3.0])
+    dfr = DataFrameRow(df, 1)
+    @test Vector(dfr) == Real[1, 2.0]
+    @test eltype(Vector(dfr)) == Real
+    @test Vector{Int}(dfr) == [1, 2]
+    @test eltype(Vector{Int}(dfr)) == Int
```
Comment (nalimilan) [inline test/dataframerow.jl]: Also test `convert` directly.

**Comment by nalimilan** [comment]
Actually, I'm a bit hesitant about whether `convert` should return a `Vector` or a `Matrix`. `Vector` sounds more logical if we consider `DataFrameRow` as equivalent to a (named) tuple, but tuples don't support conversion to vectors. `Matrix` is more consistent if we consider that `vcat(convert(Array, DataFrameRow(df, 1)), convert(Array, DataFrameRow(df, 2)))` should be equivalent to `convert(Array, df[1:2,:])`.

File: src/dataframerow/dataframerow.jl (hunk 1 above)
Comment (bkamins) [reply src/dataframerow/dataframerow.jl]: Interesting - I did not think earlier abut such case. Where is this documented?

**Comment by bkamins** [comment]
OK - we can leave it as is.

File: test/dataframerow.jl (hunk 2 above)
Comment (bkamins) [reply test/dataframerow.jl]: added

File: src/dataframerow/dataframerow.jl (hunk 1 above)
Comment (nalimilan) [reply src/dataframerow/dataframerow.jl]: Well, you never need to qualify calls to exported functions from other modules. You don't write `Base.length` either, do you? :-)

Comment (nalimilan) [reply src/dataframerow/dataframerow.jl]: Also, the CI failure is due to `df` instead of `dfr`.

Comment (bkamins) [reply src/dataframerow/dataframerow.jl]: Sure - I thought you wanted to strip `Base.` from `Base.Vector` and got confused.

**Pull request merged**